qubesadmin/tools/qvm_template.py

```diff
@@ -0,0 +1,1269 @@
+                        and cmp_res != 1:
+                    print(("Template '%s' of higher version"
+                        " already installed, skipping..." % name),
+                        file=sys.stderr)
+                    return
+
+            verified_rpm_list.append((rpmfile, reponame, name, package_hdr))
+
+        # Process local templates
+        for template in args.templates:
+            if template.endswith('.rpm'):
+                if not os.path.exists(template):
+                    parser.error('RPM file \'%s\' not found.' % template)
+                unverified_rpm_list.append((template, '@commandline'))
+
+        # First verify local RPMs and extract header
+        for rpmfile, reponame in unverified_rpm_list:
+            verify(rpmfile, reponame)
+        unverified_rpm_list = []
+
+        os.makedirs(args.cachedir, exist_ok=True)
+
+        # Get list of templates to download
+        dl_list = get_dl_list(args, app, version_selector=version_selector)
+        dl_list_copy = dl_list.copy()
+        for name, entry in dl_list.items():
+            # Should be ensured by checks in repoquery
+            assert entry.reponame != '@commandline'
+            # Verify that the templates to be downloaded are not yet installed
+            # Note that we *still* have to do this again in verify() for
+            # already-downloaded templates
+            if not override_existing and name in app.domains:
+                print(('Template \'%s\' already installed, skipping...'
+                    ' (You may want to use the'
+                    ' {reinstall,upgrade,downgrade}'
+                    ' operations.)') % name, file=sys.stderr)
+                del dl_list_copy[name]
+            else:
+                # XXX: Perhaps this is better returned by download()
+                version_str = build_version_str(entry.evr)
+                target_file = \
+                    '%s%s-%s.rpm' % (PACKAGE_NAME_PREFIX, name, version_str)
+                unverified_rpm_list.append(
+                    (os.path.join(args.cachedir, target_file), entry.reponame))
+        dl_list = dl_list_copy
+
+        # Ask the user for confirmation before we actually download stuff
+        if override_existing and not args.yes:
+            override_tpls = []
+            # Local templates, already verified
+            for _, _, name, _ in verified_rpm_list:
+                override_tpls.append(name)
+            # Templates not yet downloaded
+            for name in dl_list:
+                override_tpls.append(name)
+
+            print('This will override changes made in the following VMs:',
+                file=sys.stderr)
+            for name in override_tpls:
+                print('  %s' % name, file=sys.stderr)
+            confirm = ''
+            while confirm != 'y':
+                confirm = input('Are you sure? [y/N] ').lower()
+                if confirm == 'n':
```

Comment:
This definitely should be `confirm != 'y'`.

Reply:
The idea was to ask again if the input is neither 'y' or 'n', hence the while loop. AFAIK some CLI tools do this.

(I'm also okay with making it ask only once for simplicity though.)